Chart/assets/js/utils.js

```diff
@@ -0,0 +1,173 @@
+const button2 = document.getElementById('hour');
+const button3 = document.getElementById('day');
+const button4 = document.getElementById('week');
+const button5 = document.getElementById('month');
+button1.addEventListener('click', function() {
+    console.log('quarter was clicked!');
+    let quarter = new Date(now1);
+    quarter.setMinutes(now1.getMinutes()-15);
+    start_date=formatDate(quarter);
+    console.log(start_date);
+    end_date=formattedDate;
+    console.log(end_date);
+
+    interval='minute';
+    period=1;
+    time_series();
+});
+button2.addEventListener('click', function() {
+    console.log('1 Hour was clicked!');
+    let hour = new Date(now1);
+    hour.setMinutes(now1.getMinutes()-60);
+    start_date=formatDate(hour);
+    console.log(start_date);
+    end_date=formattedDate;
+    console.log(end_date);
+
+    interval='minute';
+    period=5;
+    time_series();
+});
+button3.addEventListener('click', function() {
+    console.log('1 Day was clicked!');
+    let day = new Date(now1);
+    day.setHours(now1.getHours()-24);
+    start_date=formatDate(day);
+    console.log(start_date);
+    end_date=formattedDate;
+    console.log(end_date);
+
+    interval='hourly';
+    period=1;
+    time_series();
+});
+button4.addEventListener('click', function() {
+    console.log('1 Week was clicked!');
+    let week = new Date(now1);
+    week.setHours(now1.getHours()-24*7);
+    start_date=formatDate(week);
+    console.log(start_date);
+    end_date=formattedDate;
+    console.log(end_date);
+
+    interval='hourly';
+    period=6;
+    time_series();
+});
+button5.addEventListener('click', function() {
+    console.log('1 Month was clicked!');
+    let month = new Date(now1);
+    month.setMonth(now1.getMonth()-1);
+    start_date=formatDate(month);
+    console.log(start_date);
+    end_date=formattedDate;
+    console.log(end_date);
+
+    interval='daily';
+    period=1;
+time_series();
+});
+
+
+function time_series(){
+    axios.get(`https://marketdata.tradermade.com/api/v1/timeseries?api_key=IhNDplgSDJWAxAXFyIKH&currency=${Exchange}&format=columns&start_date=${start_date}&end_date=${end_date}&interval=${interval}&period=${period}`)
+
+            .then(response => {
+            const Numbers = localStorage.setItem('Data',JSON.stringify(response.data));  
+            let Data=response.data.quotes.close;
+            let Data_points=Object.values(Data);
+            console.log(Data_points);
+            sum = Data_points.reduce((accumulator, current) => accumulator + current, 0);
+            Average(Data_points);
+            average_div.textContent='$' + average;
+            Rate(Data_points);
+            rate_div.textContent=rate;
+            Percent(Data_points);
+            percent_div.textContent=`(${percent}%)`;
+            chart_points(Data_points);
+            })
+            .catch(error => {
+            console.log(error);
+            });
+}
+function Average(array){
+average=sum/array.length;
```

Comment:
check implementation because all displayed values are not correct (-values)

Reply:
I rechecked it again and the values are almost the same in currency converter